chainer/iterators/order_samplers.py

```diff
@@ -0,0 +1,46 @@
+import numpy
+
+
+class ShuffleOrderSampler(object):
+
+    def __init__(self, seed=None):
+        # Use a distinct RandomState in the thread
+        # for deterministic random number generation.
+        # To support 32-bit platform and numpy < 1.11,
+        # the seed is taken in a verbose manner.
+        if seed is None:
+            seed = numpy.asscalar(
+                numpy.random.randint(-(1 << 31), 1 << 31, 1).astype('uint32'))
+        self._random = numpy.random.RandomState(seed)
+
+    def __call__(self, current_order, current_position):
```

Comment:
Do you have an example on how `current_position` and anything but the length of `current_order` is going to be used?